UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -124,6 +128,28 @@ private void OnControllerColliderHit(ControllerColliderHit hit)
         }
     }
 
+    private void CheckUnderneathStabilityAndSlide()

```

Comment:
I think your method will fail when the character stands at the end of an edge like this
![image](https://user-images.githubusercontent.com/33123731/95008300-3610bf80-0636-11eb-9ad1-9500888c4944.png)

Reply:
Does the character controller isGrounded returns true in your screenshot?
For me, the case you described, the character should be falling from the normal gravity and not from the slide.

Reply:
yes it does return true.
Yes the character should be falling but the capsule collider of the player is touching the wall at the edge and the charactercontroller thinks that player is on the ground as a result. 
I think the code you have written checks the angle with the actual ground rather then the wall here as you are casting a ray directly to the downwards from the center of the player.

Reply:
Thanks for the feedback. I was not aware of the exact behavior of isGrounded.
I changed my raycast into a SphereCastAll. It should catch the situation you described. I also saw #58 and tested it. It should fix it. Perhaps you could test it and confirm it?

Reply:
Yes this did fix the problem.😊

Reply:
Thanks for confirming it and the feedback!